### PR TITLE
[ANCHOR-832] Add Kafka `PLAINTEXT`, `SASL_PLAINTEXT` and `SASL_SSL` mode integration tests

### PR DESCRIPTION
**Pull request by lijamie98**
### Description

- Add Kafka integration tests for PLAINTEXT, SASL_PLAINTEXT, and SASL_SSL protocols.

### Context

- Improve reliability.

### Testing

- `./gradlew test`

### Documentation

N/A

### Known limitations

N/A

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -144,8 +144,10 @@ subprojects {
         val existingFile = file("$buildDir/resources/main/metadata.properties")
         existingFile.delete()
       }
-      // This is to replace the %APP_VERSION_TOKEN% in the  metadata.properties file.
-      filter { line -> line.replace("%APP_VERSION_TOKEN%", rootProject.version.toString()) }
+      filesMatching("**/metadata.properties") {
```
Comment (lijamie98) [inline build.gradle.kts:147]: without matching, the `jks` files are filtered and modified which causes the tests to fail.

File: service-runner/src/main/resources/common/docker-compose.yaml [hunk 2]
```diff
@@ -47,21 +47,46 @@ services:
     ports:
       - "29092:29092"
       - "29093:29093"
+      - "29094:29094"
+      - "29095:29095"
+      - "29096:29096"
+      - "29097:29097"
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply service-runner/src/main/resources/common/docker-compose.yaml:53]: They are listening ports and will not increase the system requirements or taking extra system resources.

Is there any downside of having them in the commons?

File: essential-tests/src/testFixtures/kotlin/org/stellar/anchor/platform/integrationtest/KafkaTests.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import org.stellar.anchor.platform.config.KafkaConfig
+import org.stellar.anchor.platform.config.KafkaConfig.SaslMechanism.PLAIN
+import org.stellar.anchor.platform.config.KafkaConfig.SecurityProtocol.*
+import org.stellar.anchor.platform.config.PropertySecretConfig.*
+import org.stellar.anchor.platform.configurator.SecretManager
+import org.stellar.anchor.platform.event.KafkaSession
+
+@ExtendWith(LockAndMockTest::class)
+@Execution(SAME_THREAD)
+class KafkaTests {
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply essential-tests/src/testFixtures/kotlin/org/stellar/anchor/platform/integrationtest/KafkaTests.kt:29]: Yeah, I am also thinking the same. Let me move.

File: service-runner/src/main/resources/common/docker-compose.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipliu) [reply service-runner/src/main/resources/common/docker-compose.yaml:53]: Ah, I didn't realize that's what it was doing. Never mind!

**Review by philipliu — APPROVED** [review]

File: service-runner/build.gradle.kts [hunk 4]
```diff
@@ -41,6 +41,8 @@ tasks { bootJar { archiveBaseName.set("anchor-platform-runner") } }
 
 application { mainClass.set("org.stellar.anchor.platform.ServiceRunner") }
 
+
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply service-runner/build.gradle.kts]: removed.

**Pull request merged**